isisdaeApp/src/isisdaeDriver.cpp

```diff
@@ -1430,6 +1430,11 @@ void isisdaeDriver::updateRunStatus()
         {
             m_RunStatus = rs;
         }
+        long run_number = m_iface->getRunNumber(); // for when we end a run
+        setIntegerParam(P_IRunNumber, run_number);
+        char run_number_str[16];
+        epicsSnprintf(run_number_str, sizeof(run_number_str), "%08ld", run_number);
```

Comment:
Are there any instruments still on 5-digit runnumbers? Like GEM for example?

Reply:
fixed in a slightly hacky way - got length from slower string read as no convenient function